app/models/operatingsystems/redhat.rb

```diff
@@ -25,9 +25,23 @@ def pxe_type
     "kickstart"
   end
 
+  def pxe_file_names(medium_provider)
+    if medium_provider&.architecture_name&.match?(/^[Ss]390/)
+      {
+        kernel: "kernel.img",
+        initrd: "initrd.img",
+      }
+    else
+      family.constantize::PXEFILES
```

Comment:
Let's fall back to the default implementation
```suggestion
      super
```